android/src/main/java/com/infitio/adharasocketio/AdharaSocketIoPlugin.java

```diff
@@ -34,21 +35,36 @@ private void log(Object message) {
     }
   }
 
-  private AdharaSocketIoPlugin(Registrar registrar, StreamsChannel streamsChannel) {
+  public AdharaSocketIoPlugin() {}
+
+  private AdharaSocketIoPlugin(FlutterPluginBinding flutterPluginBinding, StreamsChannel streamsChannel) {
     this.instances = new SparseArray<>();
     this.currentIndex = 0;
-    this.registrar = registrar;
+    this.flutterPluginBinding = flutterPluginBinding;
     this.streamsChannel = streamsChannel;
     setupStreamsChannel();
   }
 
-  public static void registerWith(Registrar registrar) {
-    final StreamsChannel streamsChannel = new StreamsChannel(registrar.messenger(), PlatformConstants.MethodChannelNames.streamsChannel);
-    final AdharaSocketIoPlugin plugin = new AdharaSocketIoPlugin(registrar, streamsChannel);
-    final MethodChannel channel = new MethodChannel(registrar.messenger(), PlatformConstants.MethodChannelNames.managerMethodChannel);
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
+    final StreamsChannel streamsChannel = new StreamsChannel(flutterPluginBinding.getBinaryMessenger(), PlatformConstants.MethodChannelNames.streamsChannel);
+    final AdharaSocketIoPlugin plugin = new AdharaSocketIoPlugin(flutterPluginBinding, streamsChannel);
+    final MethodChannel channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), PlatformConstants.MethodChannelNames.managerMethodChannel);
     channel.setMethodCallHandler(plugin);
   }
 
+  @Override
+  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
+  }
+
+
+//  public static void registerWith(Registrar registrar) {
+//    final StreamsChannel streamsChannel = new StreamsChannel(registrar.messenger(), PlatformConstants.MethodChannelNames.streamsChannel);
+//    final AdharaSocketIoPlugin plugin = new AdharaSocketIoPlugin(registrar, streamsChannel);
+//    final MethodChannel channel = new MethodChannel(registrar.messenger(), PlatformConstants.MethodChannelNames.managerMethodChannel);
+//    channel.setMethodCallHandler(plugin);
+//  }
```

Comment:
It would be good to avoid unwanted commented code.

Reply:
I could not agree more. I will remove.